component/service/src/main/java/org/exoplatform/social/rest/api/ProfileSettingsRest.java

```diff
@@ -0,0 +1,198 @@
+                                                , required = true) ProfilePropertySettingEntity profilePropertySettingEntity) {
+    long currentIdentityId = RestUtils.getCurrentUserIdentityId();
+    if (currentIdentityId == 0) {
+      return Response.status(Response.Status.UNAUTHORIZED).build();
+    }
+    if (profilePropertySettingEntity == null || profilePropertySettingEntity.getPropertyName() == null) {
+      return Response.status(Status.BAD_REQUEST).entity("Profile property setting is null or property name not provided").build();
+    }
+    try {
+      ProfilePropertySetting newProfilePropertySetting = profilePropertySettingsService.createPropertySetting(EntityBuilder.buildProfilePropertySettingFromEntity(profilePropertySettingEntity));
+      if (!profilePropertySettingEntity.getLabels().isEmpty()) {
+        labelService.createLabels(profilePropertySettingEntity.getLabels(), ProfilePropertySettingsService.LABELS_OBJECT_TYPE, String.valueOf(newProfilePropertySetting.getId()));
+      }
+      return Response.ok().build();
+    } catch (ObjectAlreadyExistsException ex) {
+      LOG.warn("property.already.exist", ex);
+      return Response.status(Status.CONFLICT).build();
+    }catch (Exception ex) {
+      LOG.warn("Failed to create a new Property setting", ex);
+      return Response.status(HTTPStatus.INTERNAL_ERROR).build();
+    }
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @PUT
+  @RolesAllowed("administrators")
+  @Operation(
+          summary = "Update a Profile property setting",
+          method = "PUT",
+          description = "\"update a Profile property setting.")
+  @ApiResponses(value = {
+          @ApiResponse(responseCode = "200", description = "Request fulfilled"),
+          @ApiResponse(responseCode = "500", description = "Internal server error"),
+          @ApiResponse(responseCode = "400", description = "Invalid query input"),
+          @ApiResponse(responseCode = "401", description = "Unauthorized operation")})
+  public Response updatePropertySetting(@Context UriInfo uriInfo,
+                                        @RequestBody(description = "Profile property setting object to be updated"
+                                                , required = true) ProfilePropertySettingEntity profilePropertySettingEntity) {
+    long currentIdentityId = RestUtils.getCurrentUserIdentityId();
+    if (currentIdentityId == 0) {
+      return Response.status(Response.Status.UNAUTHORIZED).build();
+    }
+    if (profilePropertySettingEntity == null) {
+      return Response.status(Status.BAD_REQUEST).entity("Profile property setting is null").build();
+    }
+    try {
+      profilePropertySettingsService.updatePropertySetting(EntityBuilder.buildProfilePropertySettingFromEntity(profilePropertySettingEntity));
+      labelService.mergeLabels(profilePropertySettingEntity.getLabels(), ProfilePropertySettingsService.LABELS_OBJECT_TYPE, String.valueOf(profilePropertySettingEntity.getId()));
```

Comment:
to split to multiple instructions to make the code readable. Besides is there a business logic here ? Thus to move to Service, except for the REST DTO transformation to Service DTO.